os/kernel/log_dump/log_dump.c

```diff
@@ -505,6 +505,26 @@ int log_dump_save(char ch)
 	return LOG_DUMP_OK;
 }
 
+size_t log_dump_get_size(void)
+{
+	size_t size = 0;
+	struct log_dump_chunk_s* head = (struct log_dump_chunk_s *)sq_peek(&log_dump_chunks);
+	while (head != (struct log_dump_chunk_s *)sq_tail(&log_dump_chunks)) {
+		head = (struct log_dump_chunk_s *)sq_next((sq_entry_t *)head);
+		size += CONFIG_LOG_DUMP_CHUNK_SIZE;
+	}
+	/* compress_curbytes is the length of compressed logs in latest log_dump_chunk */
+	size += compress_curbytes;
+
+	/* This is the size of compressed partially filled uncomp_buf.
+	 * Note: It is recommended to use this log_dump_get_size api after
+	 * log_dump_read_wake() is performed, so that it has correct value of last_comp_block_size.
```

Comment:
App will use this API and App can't know whether `log_dump_read_wake` is performed or not. This recommendation looks redundant. Is there any other way to do it automatically?